### PR TITLE
Add redis cache support

### LINKED ISSUE #48: Support Redis

### PR DESCRIPTION
**Pull request by atomfrede**
Uses redisson and jcache integration.

* [x] Add redis via testcontainers to integration tests
* [x] Add (at least) one ci job for redis configuration
* [x] Heroku Support
* [x] Customize needle api to support redis cache entries

closes #48 

image

### CODE REVIEW

**Comment by atomfrede** [comment]
Need to adapt the integration test to use testcontainers, but apart from that it looks fine.

**Comment by atomfrede** [comment]
I had to adapt the `JHipsterProperties` as it was not able to overwrite the properties prefixed with `jhipster`

File: generators/client/files.js [hunk 1]
```diff
@@ -59,6 +59,12 @@ const clientFiles = {
                     method: 'processJs',
                     renameTo: () => `${ANGULAR_DIR}core/login/login.service.ts`,
                 },
+                // profiles response returns a hyphenated key
+                {
+                    file: 'angular/src/main/webapp/app/layouts/profiles/profile.service.ts',
+                    method: 'processJs',
+                    renameTo: () => `${ANGULAR_DIR}layouts/profiles/profile.service.ts`,
```
Comment (atomfrede) [inline generators/client/files.js:66]: This is just something I noticed when deploying to heroku/using prod profile. For react there is already a custom reducer handling the hyphenated profile names. For angular there was not and therefore e.g. the database menu item was displayed.

File: generators/server/templates/src/main/java/package/util/JHipsterProperties.java.ejs [hunk 2]
```diff
@@ -5,10 +5,10 @@ import io.micronaut.context.annotation.ConfigurationProperties;
 @ConfigurationProperties("jhipster")
 public class JHipsterProperties {
 
-    private final Cache cache;
-    private final Mail mail;
-    private final Logging logging;
-    private final Metrics metrics;
+    private Cache cache;
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/util/JHipsterProperties.java.ejs:8]: When these where final the values could not be overwritten via properties files.

File: generators/server/templates/src/main/resources/application-test.yml.ejs [hunk 3]
```diff
@@ -46,7 +46,7 @@ jhipster:
     use-json-format: true # By default, logs are in Json format
     # To test logstash appender
     logstash:
-      enabled: true
+      enabled: false
```
Comment (atomfrede) [inline generators/server/templates/src/main/resources/application-test.yml.ejs:49]: As the values from properties are now used these should be false.

**Comment by atomfrede** [comment]
Need to customize the needle for adding a cache entry. Kotlin as an example how to do it: https://github.com/jhipster/jhipster-kotlin/blob/master/generators/entity-server/needle-server-cache-kt.js

**Comment by JasonTypesCodes** [comment]
@atomfrede Redis is a bit out of my zone.  Do you feel this is ready to merge?

**Comment by atomfrede** [comment]
I tested it locally and with heroku redis. We have basically the same config as we have for spring boot so from my side let's merge it!

**Pull request merged**